alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/healthprocessor/plugins/solr/MutableHealthReport.java

```diff
@@ -0,0 +1,76 @@
+package eu.xenit.alfresco.healthprocessor.plugins.solr;
+
+import eu.xenit.alfresco.healthprocessor.reporter.api.NodeHealthReport;
+import eu.xenit.alfresco.healthprocessor.reporter.api.NodeHealthStatus;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.NonNull;
+import org.alfresco.service.cmr.repository.NodeRef;
+
+/**
+ * Health report that is initially unset and can be marked as healthy, unhealthy or unknown for a reason
+ */
+public class MutableHealthReport {
+
+    private final NodeRef nodeRef;
+
+    @Nullable
+    private NodeHealthStatus healthStatus = null;
+    private NodeHealthStatus forcedHealthStatus = null;
+
+    private Set<String> messages = new HashSet<>();
```

Comment:
This is not the health report, but a wrapper. The actual health report is more generic and used across all health processor plugins. It takes a `Set<String>` for messages, so there is no order in the messages there anyways.

> marking an unhealthy node as healthy adds a message to the set but you don't now that the message resulted from a "markHealthy" function call

Did you see what the message are used for? They are added [here](https://github.com/xenit-eu/alfresco-health-processor/pull/34/files#diff-c5b64f8fb722c67171947463b301b220f3afecaf2030b41476e2ad83396db96aR89-R104), with one message per core that should contain the node.

So, the order of the messages does not really matter, as it would just depend on which solr core is checked first.

It is not marked as healthy after it is marked "unhealthy", because we want it to appear unhealthy if it is missing in *any* core.

Reply:
_It is not marked as healthy after it is marked "unhealthy", because we want it to appear unhealthy if it is missing in any core._ 
Yes i am aware of that. I wast just thinking it could be useful to filter on it or immediately see the origin why the message was added.

Reply:
I don't think so. Every message is only actually being added in one place.

And we currently don't have any reporters that log anything but the unhealthy node and the messages.

It is possible that we will have to look into a structured message format if we end up with some "fixer" plugins (next to the existing healthchecking plugins and reporting plugins).